.github/workflows/sanitizer-tests.yaml

```diff
@@ -90,23 +90,18 @@ jobs:
       - name: Install apt packages
         uses: ./src/Fast-DDS/.github/actions/install-apt-packages
 
-      - name: Install GTest
-        uses: ./src/Fast-DDS/.github/actions/install-gtest-linux
-
       - name: Install Python packages
         uses: ./src/Fast-DDS/.github/actions/install-python-packages
 
-      - name: Fetch repositories
+      - name: Fetch Fast DDS dependencies
+        uses: ./src/Fast-DDS/.github/actions/fetch-fastdds-repos
         env:
           DEFAULT_DISCOVERY_SERVER_BRANCH: 'master'
-        run: |
-          mkdir -p src
-          cd src
-          git clone https://github.com/eProsima/foonathan_memory_vendor.git
-          git clone https://github.com/eProsima/Fast-CDR.git
-          git clone https://github.com/eProsima/Discovery-Server.git --branch ${{ github.event.inputs.discovery_server_branch || env.DEFAULT_DISCOVERY_SERVER_BRANCH }}
-          git clone https://github.com/google/googletest.git --branch release-1.10.0
-          cd ..
+        with:
+          target_directory: src
+          googletest: true
+          discovery_server: true
+          discovery_server_branch: ${{ github.event.inputs.discovery_server_branch || env.DEFAULT_DISCOVERY_SERVER_BRANCH }}
```

Comment:
I see we use the specified version in case the workflow is manually dispatched, but if it's not, then I don't think we should always use master of Discovery Server. I think we should:

1. Use a branch with the same name as the PR target if exists; else
2. Use a branch with the same name as the PR base if exists; else
3. Use master